src/clj/rems/api/services/form.clj

```diff
@@ -133,9 +121,11 @@
 
 (defn set-form-archived! [{:keys [id archived]}]
   (util/check-allowed-organization! (:form/organization (get-form-template id)))
-  (let [catalogue-items (catalogue-items-for-form id false)]
-    (or (and archived (form-in-use-error id false))
-        (do
-          (db/set-form-template-archived! {:id id
-                                           :archived archived})
-          {:success true}))))
+  (if-let [errors (when archived
+                    (dependencies/archive-errors :t.administration.errors/form-in-use {:form/id id}))]
```

Comment:
I don't understand why the error key needs to be sent here.

Reply:
This refactoring kept the errors as-is, and the keys were different in different service files...

Reply:
Maybe the other part is that the merge with the `:type key` could happen here and not in `archive-errors`?